src/components/Album.js

```diff
@@ -30,82 +35,332 @@ import AddToWishlistButton from './layout/AddToWishlistButton/AddToWishlistButto
-              className="album-image"
-            />
-          </Button>
-          <p className="album-title">{album.albumName}</p>
-          <p className="album-artist">{album.artistName}</p>
-          <div className="button-container">
-            {/* <Button variant="contained" color="primary" className="buy-button">
-            Buy
-          </Button> */}
-            {/*AddToCartButton component */}
-            <AddToCartButton album={album} />
-
-            {/* <Button
+        <Tooltip title={album.albumName} arrow>
+          <AlbumTitle className="album-title">{album.albumName}</AlbumTitle>
+        </Tooltip>
+        <Typography className="album-artist">{album.artistName}</Typography>
+        <Box className="button-container">
+          <AddToCartButton album={album} />
+          <Button
             variant="contained"
             color="secondary"
             className="wishlist-button"
           >
             Wishlist
-          </Button> */}
-            {/*AddToCartButton component */}
-            <AddToWishlistButton album={album} />
-          </div>
-        </div>
-
-        {/* Dialog */}
-        <Dialog open={openDialog} onClose={handleCloseDialog}>
-          <DialogTitle>{album.albumName}</DialogTitle>
+          </Button>
+        </Box>
+        <Dialog
+          open={openDialog}
+          onClose={handleCloseDialog}
+          className="album-modal-main"
+        >
+          <DialogTitle
+            sx={{
+              display: 'flex',
+              justifyContent: 'space-between',
+              alignItems: 'center',
+            }}
+          >
+            {album.albumName}
+            <Box sx={{ cursor: 'pointer' }} onClick={handleCloseDialog}>
+              <CloseIcon />
+            </Box>
+          </DialogTitle>
           <DialogContent>
             <DialogContentText>
               {/* You can customize the content here */}
-              <div>
-                <div>
-                  Release Date:{' '}
-                  {new Date(album.releaseDate).toLocaleDateString()}
-                </div>
-                <div>Average Rating: {album.averageRating}</div>
-              </div>
+              <Box className="card-design-main">
+                <Box className="card-design">
+                  <img
+                    src={album.image}
+                    alt={album.albumName}
+                    className="album-image"
+                  />
+                  <Box>
+                    <Box>
+                      Release Date:{' '}
+                      {new Date(album.releaseDate).toLocaleDateString()}
+                    </Box>
+                    <Box>Average Rating: {album.averageRating}</Box>
+                  </Box>
+                </Box>
+
+                <Typography className="title" variant="h5">
+                  Review:
+                </Typography>
+                {reviewList.map((val) => (
+                  <div className="rate-card">
```

Comment:
Anytime we iterate over a list in React we have to make sure the first child has the `key` attribute. This allows React to better know how to make render updates when one of the items in the list has a state change. [https://react.dev/learn/rendering-lists](https://react.dev/learn/rendering-lists)

```suggestion
                {reviewList.map((val) => (
                  <div className="rate-card" key={val._id}>
```